lang/py/src/avro/io.py

```diff
@@ -119,11 +120,27 @@ def validate(expected_schema, datum):
       return isinstance(datum, Decimal)
     return isinstance(datum, str)
   elif schema_type == 'int':
-    return ((isinstance(datum, int) or isinstance(datum, long)) 
-            and INT_MIN_VALUE <= datum <= INT_MAX_VALUE)
+    if hasattr(expected_schema, 'logical_type'):
+      if expected_schema.logical_type == 'date':
+        return isinstance(datum, datetime.date)
+      elif expected_schema.logical_type == 'time-millis':
+        return isinstance(datum, datetime.time)
+      else:
+        return False
```

Comment:
I suspect for unknown logical type, it should ignore and fall back underlying Avro type according to the spec. Could you check how the Java implementation handles such case?